akd/src/errors.rs

```diff
@@ -158,57 +169,47 @@ impl fmt::Display for AzksError {
             Self::VerifyAppendOnlyProof => {
                 write!(f, "Append only proof did not verify!")
             }
+            Self::NoEpochGiven => {
+                write!(f, "An epoch was required but not supplied")
+            }
         }
     }
 }
 
 /// The errors thrown by various algorithms in [crate::directory::Directory]
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
 pub enum DirectoryError {
-    /// Looked up a user not in the directory
-    NonExistentUser(String, u64),
     /// Lookup proof did not verify
     VerifyLookupProof(String),
     /// Key-History proof did not verify
     VerifyKeyHistoryProof(String),
     /// Tried to audit an invalid epoch range
     InvalidEpoch(String),
-    /// Error propagation
-    Storage(StorageError),
-    /// Error propagation for errors from VRF storage
-    VRFStorageErr(VRFStorageError),
+    /// AZKS not found in read-only directory mode
```

Comment:
Technically this error is thrown when we try to perform a write operation in read-only mode, right? This includes trying to publish. I think the error message can be clearer, especially in terms of what the bool stands for.